src/code/z_play.c

```diff
@@ -1611,7 +1611,10 @@ void Play_CopyCamera(PlayState* this, s16 destCamId, s16 srcCamId) {
     Camera_Copy(this->cameraPtrs[destCamId1], this->cameraPtrs[srcCamId2]);
 }
 
-s32 func_800C0808(PlayState* this, s16 camId, Player* player, s16 setting) {
+/**
+ * Initializes the CamPlayer settings and changes the camera setting
+ */
+s32 Play_InitCameraSettings(PlayState* this, s16 camId, Player* player, s16 setting) {
```

Comment:
What is `CamPlayer` ?

also I'd like if we could avoid "settings" in camera context since it's very easy to confuse with "setting" (as in `cam_set_`)
but I'm not sure what to name this so it's fine I guess

I guess `Play_InitCamera` is too generic,
could more be taken from `Camera_InitPlayerSettings` like `Play_InitCameraPlayerSettings` ?

Reply:
I was trying to avoid `Player` in the name since I didn't want `Play_InitCameraPlayerSettings` to be interpreted as "set camera and player settings" since this is a play function, but perhaps it's fine

Reply:
tbh I'm not sure why `Camera_InitPlayerSettings` is named that either

Reply:
On a surface level, it does seem many camera values are set based on `player`, although some parts of it aren't. I do think this `z_play` function should be named after `Camera_InitPlayerSettings` in some way as that seems to be an important part of this function, but if `Camera_InitPlayerSettings`'s name needs reevaluating, then it might be worth unnaming this function for now

Reply:
(Or even better, find a good name for `Camera_InitPlayerSettings` in this PR if we want to find a better name)

Reply:
I don't have any good idea

I don't think the "player" part of `Camera_InitPlayerSettings` is super relevant since `Play_InitCameraSettings` has 100% cutscene usage, but idk.
I still wish there was no "settings" but no idea

Can you change `CamPlayer` at least? idk maybe "camera player settings" instead (of "CamPlayer settings")
(it looks like a struct name as-is)

Reply:
Also, very specifically onepointdemo with one exception, where child zelda first notices you in the courtyard. Could lean into the cutscene part of it. Thinking of it from a data point of view:

`Camera_InitPlayerSettings` -> `Camera_InitData`?
`Play_InitCameraSettings` -> `Play_InitCameraData`? (thinking of your KeepOn4 PR where there's the generic `data` struct set with `Camera_SetCameraData`)

Reply:
Ended up with `Camera_InitDataWithPlayer` and `Play_InitCameraDataWithPlayer` for these two functions

Reply:
Out of curiosity why add the `WithPlayer`?
On first glance it almost seems like it might be initializing `Player` somehow (though this is blatantly wrong and perhaps just me). If it doesn't conflict with other names, I feel like the `Player` arg to the function kind of explains it will be initializing camera data based on `Player` somehow.

Reply:
I think my current interpretation is that `Camera_InitData` alone makes it sound like it's needed to initialize data when it's not. This appears to be an optional function to initialize camera data based around player-related variables. You're right in that it doesn't actually change anything onto the Player itself, more that it uses those player-related variables

Reply:
Perhaps `SetData`? or maybe `SetPlayerData`?

Reply:
It is generally a function that runs when initializing a camera, as you'd want to leave any changes afterwards up to the camera update functions. The `PlayerData` -> `DataWithPlayer` was originally meant to help clarify that this is setting data for camera, and that player is just an add-on/support to guide how those data is set 😅

Reply:
Sounds good. This isn't a deal breaker, so I'll approve.

Reply:
not sure if we can try to further rephrased as like `PlayerRelatedData`
cause I do agree that "with player" can pretty easily be interpreted as player is being initialized with the camera. but also dont think its a huge deal that needs to be settled right now

Reply:
To try give an idea of when this is called:

1) an actor calls `OnePointCutscene_Init` to start an """attention cutscene"""
2) `Camera_Init` is called on a new subcam: `OnePointCutscene_Init` > `Play_CreateSubCamera` > `Camera_Init`
3) `OnePointCutscene_SetInfo` is called by `OnePointCutscene_Init`
4) Many (but not all) `csId`s call `func_800C0808`/`Play_InitCameraDataWithPlayer` which in turn calls `Camera_InitPlayerSettings`/`Camera_InitDataWithPlayer`

Additionally:
- EnZl1 has one use of `func_800C0808`/`Play_InitCameraDataWithPlayer`
- `Camera_InitPlayerSettings`/`Camera_InitDataWithPlayer` is called in `Play_Init`

Note in all cases `Camera_Init`, the "real" camera_init, has already run,
so the functions we're talking about `func_800C0808`/`Play_InitCameraDataWithPlayer` and `Camera_InitPlayerSettings`/`Camera_InitDataWithPlayer`
are to me "further initialize" rather than initialize in full, which I think "InitDataWithPlayer" conveys

Not that I think everything in `Camera_InitPlayerSettings`/`Camera_InitDataWithPlayer` makes sense for its purpose/usage either, hence why a vague name ("data") seems appropriate

Reply:
To clarify, what is currently called an "attention cutscene" is 1 specific case of a OnePoint Cutscene, specificly csId = 5010 which instead of initializing its keyframes in the giant OnePoint switch-case, instead initializes its keyframes in `Camera_Demo5` using `CAM_SET_CS_ATTENTION`, and all these onePoint functions are for this specific type:
https://github.com/zeldaret/oot/blob/b9d746aba55109f4e28445dbb89b6b5013296b8f/src/code/z_onepointdemo.c#L1337-L1469

So a onePoint "attention cutscene" does use `Play_InitCameraDataWithPlayer`, but many non-attention onePoint cutscenes also use `Play_InitCameraDataWithPlayer`

Reply:
Thoughts on maybe using `BasedOn` instead of `With`?
Not sure how I feel about it myself, but does help alleviate the first impression of "are we somehow initializing player here`?

Reply:
`Camera_InitDataUsingPlayer` and `Play_InitCameraDataUsingPlayer`? Maybe?

Reply:
im also ok with this